### PR TITLE
Trial implementation of test code for debbuger -multi processes

### PR DESCRIPTION
**Pull request by ono-max**
This is the trial implementation of test code for debugger by using multi processes.

- Use module PTY to create pseudo terminals for debugger.
- Write test codes for `next` and `step` and `break`

### How to use

```
# all test will be executed
$ ruby bin/test-unit.rb

# specify test file
$ ruby bin/test-unit.rb debug/break_test.rb
```

### Demo
[Image from Gyazo](https://gyazo.com/2bbe47cb20b9a49f48a0c6c0ad146d9e)

### Reference

The following test code is based on [byebug ](https://github.com/deivid-rodriguez/byebug)project

### CODE REVIEW

File: bin/minitest [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative '../test/minitest_runner'
```
Comment (ko1) [inline bin/minitest:4]: is this file needed?

Comment (ono-max) [reply bin/minitest:4]: No, I will remove it.

Comment (ono-max) [reply bin/minitest]: f592487

Comment (ono-max) [reply bin/minitest]: Done

File: bin/test-unit.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+base_dir = File.expand_path(File.join(File.dirname(__FILE__), ".."))
+lib_dir  = File.join(base_dir, "lib")
+test_dir = File.join(base_dir, "test")
+
+$LOAD_PATH.unshift(lib_dir)
+
+require 'test/unit'
+
+exit Test::Unit::AutoRunner.run(true, test_dir)
```
Comment (ono-max) [inline bin/test-unit.rb:9]: copy and paste codes in reference
https://test-unit.github.io/test-unit/en/file.how-to.html

File: test/debug/break_test.rb [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+      debug_code(program) do
+        type 'break 3'
+        type 'continue'
+        assert_line_num 3
+        type 'quit'
+        type 'y'
+      end
+    end
+  end
+end
```
Comment (ono-max) [inline test/debug/break_test.rb:160]: I refer to codes in byebug.
https://github.com/deivid-rodriguez/byebug/blob/master/test/commands/break_test.rb

File: test/support/utils.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+          read.expect(/(.*)\n|\(rdbg\)/) do |sentence|
+            print sentence[0]
+            if sentence[0] == '(rdbg)'
+              cmd = @queue.pop
+              if cmd.is_a?(Proc)
+                cmd.call(result)
+                cmd = @queue.pop
+              end
+              write.puts(cmd)
+              quitted = false if cmd == 'quit'
```
Comment (st0012) [inline test/support/utils.rb:47]: just curious about what `quitted` means here. I thought if `quitted == true`, the terminal would stop. but it looks like it's the opposite?

Comment (ono-max) [reply test/support/utils.rb:47]: Oh, that was my mistake. Thanks!

**Comment by ko1** [comment]
@deivid-rodriguez

Hi, we made a test based on your byebug tests.
Now we write your license in a comment.
How about that? Or is it enough "thanks" comment in README.md?

Thank you!

**Comment by deivid-rodriguez** [comment]
README thanks is enough! :)

**Comment by ko1** [comment]
> README thanks is enough! :)

Thank you for your quick reply!
